### PR TITLE
sGt1

### PR DESCRIPTION
**Pull request by NoJungi**
Returns a list of IDs of the k best chunks.
I created a directory that contains some example chunks from the sEt3 branch and the corresponding hdf5 file for tests.

### CODE REVIEW

File: src/chunks_for_question.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+def get_k_IDs(question: str,
+              embeddings_file: str,
+              k: int = 5) -> list[int]:
+    """Gets the IDs of the k chunks that have the best cosine similarity with 
+    the embedded question. The embeddings of the chunks are given in the hdf5
+    file named after the string of 'embeddings_file' which should be locaded in
+    the data directory of this repository. The function gives back a list that 
+    contains the IDs of the k best chunks starting with the best match for the 
+    question.
```
Comment (MarioStanke) [inline src/chunks_for_question.py:16]:
```suggestion
    contains the IDs of the k chunks sorted decreasingly by similarity to the question.
```

File: src/chunks_for_question.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    Args:
+        question (str): The string of the question to compare the embeddings 
+            of the chunks to.
+        embeddings_file (str): The relative path of the hdf5 file, that 
+            contains the embeddings , if one is currently in the data 
+            directory, since all data files should be stored there. 
+            Example:    For the file named "example_embeddings" in the  
+                        directory "data/folder" the strings should be in the 
+                        following format: "foler/example_embeddings"
```
Comment (MarioStanke) [inline src/chunks_for_question.py:26]: folder

File: src/chunks_for_question.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            directory, since all data files should be stored there. 
+            Example:    For the file named "example_embeddings" in the  
+                        directory "data/folder" the strings should be in the 
+                        following format: "foler/example_embeddings" 
+                        (without "/" at the beginning)
+            Attention: The dictionary has to have a key that contains the 
+                       string "embeddings" that gives the array of the 
+                       embeddigs and a string with "ids" that gives the ids or
+                       names of the corrsponding files.
+        k (int): Integer that indicates the number of chunks that are returned.
```
Comment (MarioStanke) [inline src/chunks_for_question.py:32]:
```suggestion
        k (int):          the maximum number of chunks that are returned.   
```
```suggestion
        k (int): Integer that indicates the number of chunks that are returned.   
```

File: src/chunks_for_question.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                        (without "/" at the beginning)
+            Attention: The dictionary has to have a key that contains the 
+                       string "embeddings" that gives the array of the 
+                       embeddigs and a string with "ids" that gives the ids or
+                       names of the corrsponding files.
+        k (int): Integer that indicates the number of chunks that are returned.   
+
+    Returns:
+        list[int]: The list that contains the IDs of the k files with the 
+            best cosine similiarity for the question orderd from most to least
```
Comment (MarioStanke) [inline src/chunks_for_question.py:36]:
```suggestion
            best cosine similiarity for the question ordered from most to least 
```

File: src/chunks_for_question.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            similar.
+    """
+    # Check if question is given:
+    if not question:
+        print("No question was given. Please enter a question.")
+        return [None]
+
+    # Get the embeddings from the hpf5 file if exists and acess is given:
+    file_path = os.getcwd() + "/../data/" + embeddings_file
+    if(not os.path.isfile(file_path) or not os.access(file_path, os.R_OK)):
```
Comment (MarioStanke) [inline src/chunks_for_question.py:46]:
```suggestion
    if (not os.path.isfile(file_path) or not os.access(file_path, os.R_OK)):
```

File: src/chunks_for_question.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        embedding_list (list[list[float]]): The list containing the embeddings 
+            of the chunks.   
+
+    Returns:
+        list[int]: The list that contains the indices of the argsort of the 
+            embeddings according to the cosine similiarity for the question 
+            orderd from most to least similar.
+    """
+    # Embed the question
+    question_embedding = embedding_from_string(string=question)
```
Comment (MarioStanke) [inline src/chunks_for_question.py:92]: This should be outside of this function. For example, if we want to compute multiple lists sorted with different parameters then we still need to compute the embedding of the question once.

**Comment by NoJungi** [comment]
I have changed the code for the requests. It works with the example hdf5 file, we may have to test it again later with the real data.

**Review by MarioStanke — APPROVED** [review]
LGTM

**Pull request merged**